cluster/addons/metadata-proxy/gce/metadata-proxy.yaml

```diff
@@ -27,26 +27,34 @@ spec:
       dnsPolicy: Default
       containers:
       - name: metadata-proxy
-        image: gcr.io/google-containers/metadata-proxy:0.1.3
-        imagePullPolicy: Always
+        image: gcr.io/google_containers/metadata-proxy:v0.1.4
+        # TODO(#8867) Remove privileged securityContext once iptables munging
+        # is in ip-masq-agent.
         securityContext:
           privileged: true
-        command:
-          - '/start-proxy.sh'
         resources:
           requests:
             memory: "32Mi"
             cpu: "50m"
           limits:
             memory: "32Mi"
             cpu: "50m"
+      # BEGIN_PROMETHEUS_TO_SD
+      - name: prometheus-to-sd-exporter
+        image: gcr.io/google_containers/prometheus-to-sd:v0.2.2
+        command:
+          - /monitor
+          - --stackdriver-prefix={{ prometheus_to_sd_prefix }}/addons/metadata_proxy
```

Comment:
you don't need to add metadata_proxy as a prefix. Metric name is built as:
prefix/component_name/(metric_name without component_name)

Reply:
Done; how does it know the component_name?

Reply:
It's taken from the first part of --source flag. So if you specify multiple sources, each one is going to have a separate stackdriver prefix.